### ISSUE TITLE
Text Sets A11Y

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#6295]
## Bug Description

Two observations at a glance

* `TextSetItem` uses `role=listitem`, but there's no parent with `role=list`
* Every text set has the same label `Insert Text Set` which is not helpful at all
* Category selection/deselection doesn't seem to get announced to screen readers (`useLiveRegion` might be helpful here)
* To screen readers it's not clear what the list of categories does. Need to announce their purpose, perhaps via `aria-label` or by using `VisuallyHidden`

[Complete list of text sets](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=2%3A30878)

## Expected Behaviour

* List items need to be in a proper list
* Each text set should be labelled with its individual name

## Additional Context

- Plugin Version: 1.3
- WordPress Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by BrittanyIRL** [comment]
@embarks I just merged the label update for text sets which should unblock the rest of this ticket

**Comment by csossi** [comment]
Verified in QA
